cmd/k8s-operator/deploy/crds/tailscale.com_connectors.yaml

```diff
@@ -88,32 +88,46 @@ spec:
                   description: List of status conditions to indicate the status of the Connector. Known condition types are `ConnectorReady`.
                   type: array
                   items:
-                    description: ConnectorCondition contains condition information for a Connector.
+                    description: "Condition contains details for one aspect of the current state of this API Resource. --- This struct is intended for direct use as an array at the field path .status.conditions.  For example, \n type FooStatus struct{ // Represents the observations of a foo's current state. // Known .status.conditions.type are: \"Available\", \"Progressing\", and \"Degraded\" // +patchMergeKey=type // +patchStrategy=merge // +listType=map // +listMapKey=type Conditions []metav1.Condition `json:\"conditions,omitempty\" patchStrategy:\"merge\" patchMergeKey:\"type\" protobuf:\"bytes,1,rep,name=conditions\"` \n // other fields }"
```

Comment:
Nit: these comments are user facing docs (when running `kubectl explain`), so probably the struct comment should not be here, it seems confusing for users, see:
```
irbe@tailscale$ kubectl explain connector.status.conditions
GROUP:      tailscale.com
KIND:       Connector
VERSION:    v1alpha1

FIELD: conditions <[]Object>

DESCRIPTION:
    List of status conditions to indicate the status of the Connector. Known
    condition types are `ConnectorReady`.
    Condition contains details for one aspect of the current state of this API
    Resource. --- This struct is intended for direct use as an array at the
    field path .status.conditions.  For example, 
     type FooStatus struct{ // Represents the observations of a foo's current
    state. // Known .status.conditions.type are: "Available", "Progressing", and
    "Degraded" // +patchMergeKey=type // +patchStrategy=merge // +listType=map
    // +listMapKey=type Conditions []metav1.Condition
    `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"
    protobuf:"bytes,1,rep,name=conditions"` 
     // other fields }
    ....
```

Reply:
Ahh, do we actually get this from upstream?

Reply:
Ok, ignore my comment, I see that we get this from the upstream condition type. I guess there is nothing we can do.

Reply:
It looks like the `controller-gen` PR might get merged 🤞 Given this PR isn't urgent I'm inclined to wait and see for a bit. WDYT?

Reply:
> diff from that looks good to me

That's great news, thank you for following up with the controller tools PR.

The only reason to merge now would be, so that we have more time to observe how the ne Service conditions behave in unstable releases, before the becomes part of a stable release. But I would also vote for waiting to see the upstream PR get merged at least 👍🏼

Reply:
Apart from that, this looks good to go in to me 👍🏼

Reply:
Assuming CI goes green shortly, this should be ready for another look now 👀